src/DIRAC/TransformationSystem/Client/WorkflowTasks.py

```diff
@@ -460,31 +465,38 @@ def _handleHospital(self, oJob, clinicPath):
 
     def __generatePluginObject(self, plugin):
         """This simply instantiates the TaskManagerPlugin class with the relevant plugin name"""
+
         method = "__generatePluginObject"
         try:
-            plugModule = __import__(self.pluginLocation, globals(), locals(), ["TaskManagerPlugin"])
+            plugModule = importlib.import_module(self.pluginLocation)
         except ImportError as e:
-            self._logException("Failed to import 'TaskManagerPlugin' %s: %s" % (plugin, e), method=method)
+            self._logException(f"Failed to import 'TaskManagerPlugin' {plugin}: {e}", method=method)
             return S_ERROR()
         try:
             plugin_o = getattr(plugModule, "TaskManagerPlugin")("%s" % plugin, operationsHelper=self.opsH)
             return S_OK(plugin_o)
         except AttributeError as e:
-            self._logException("Failed to create %s(): %s." % (plugin, e), method=method)
+            self._logException(f"Failed to create {plugin}(): {e}", method=method)
             return S_ERROR()
 
     #############################################################################
 
     def getOutputData(self, paramDict):
         """Get the list of job output LFNs from the provided plugin"""
         if not self.outputDataModule_o:
-            # Create the module object
-            moduleFactory = ModuleFactory()
+            # TODO: check with Federico if we shouldn't use ObjectLoader every time
```

Comment:
It would make sense to use `ObjectLoader`

Reply:
Should I also use ObjectLoader for the `__generatePluginObject` method or not ?

Reply:
yes